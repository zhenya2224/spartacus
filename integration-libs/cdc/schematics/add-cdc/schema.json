{
  "$schema": "http://json-schema.org/schema",
  "$id": "CDCSchematics",
  "title": "CDC Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the CDC feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["CDC"]
    }
  },
  "required": []
}
