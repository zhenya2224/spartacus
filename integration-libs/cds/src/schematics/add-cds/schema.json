{
  "$schema": "http://json-schema.org/schema",
  "$id": "CDSSchematics",
  "title": "CDS Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["CDS"]
    },
    "tenant": {
      "type": "string",
      "description": "The tenant name.",
      "default": "my-tenant",
      "x-prompt": "[CDS] What is your tenant name?"
    },
    "baseUrl": {
      "type": "string",
      "description": "The base URL.",
      "default": "https://my-tenant.api.us.context.cloud.sap",
      "x-prompt": "[CDS] What is your base URL?"
    },
    "profileTagLoadUrl": {
      "type": "string",
      "description": "Where to load ProfileTag from. (e.g. https://tag.static.us.context.cloud.sap/js/profile-tag.js)",
      "x-prompt": "[CDS] Where to load ProfileTag from? (optional)"
    },
    "profileTagConfigUrl": {
      "type": "string",
      "description": "Configuration URL for ProfileTag. (e.g. https://tag.static.stage.context.cloud.sap/config/my-config123)",
      "x-prompt": "[CDS] What is the ProfileTag configuration URL? (optional)"
    }
  }
}
