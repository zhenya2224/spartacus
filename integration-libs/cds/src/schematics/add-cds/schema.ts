import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  tenant: string;
  baseUrl: string;
  profileTagLoadUrl?: string;
  profileTagConfigUrl?: string;
}
