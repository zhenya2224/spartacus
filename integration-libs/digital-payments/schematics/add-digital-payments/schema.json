{
  "$schema": "http://json-schema.org/schema",
  "$id": "DigitalPaymentsSchematics",
  "title": "DigitalPayments Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Digital-Payments"]
    }
  },
  "required": []
}
