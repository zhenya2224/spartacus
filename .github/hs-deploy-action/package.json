{
  "name": "deploy-action",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "build": "tsc index.ts"
  },
  "author": "",
  "license": "Apache-2.0",
  "dependencies": {
    "@actions/exec": "^1.0.4",
    "@actions/github": "^4.0.0",
    "@types/node": "^12.11.1"
  },
  "devDependencies": {
    "typescript": "^4.1.3"
  }
}
