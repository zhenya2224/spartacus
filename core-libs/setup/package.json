{
  "name": "@spartacus/setup",
  "version": "4.1.0-next.0",
  "description": "Includes features that makes Spartacus and it's setup easier and streamlined.",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "setup",
    "recipe"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/core-libs/setup",
  "license": "Apache-2.0",
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/core": "^12.0.5",
    "@spartacus/checkout": "4.1.0-next.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/user": "4.1.0-next.0"
  },
  "optionalDependencies": {
    "@nguniversal/express-engine": "^12.0.2",
    "express": "^4.15.2"
  },
  "publishConfig": {
    "access": "public"
  }
}
