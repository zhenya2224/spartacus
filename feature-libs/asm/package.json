{
  "name": "@spartacus/asm",
  "version": "4.1.0-next.0",
  "description": "ASM feature library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "asm"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/asm",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^12.0.5",
    "@angular/common": "^12.0.5",
    "@angular/core": "^12.0.5",
    "@angular/forms": "^12.0.5",
    "@ngrx/effects": "^12.1.0",
    "@ngrx/store": "^12.1.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
