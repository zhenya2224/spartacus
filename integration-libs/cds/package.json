{
  "name": "@spartacus/cds",
  "version": "4.0.0-rc.1",
  "description": "Context Driven Service library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "merchandising",
    "segmentation",
    "context-driven services",
    "cds"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cds",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"src/schematics/**/*.js\" \"src/schematics/**/*.js.map\" \"src/schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^12.0.5",
    "@angular/common": "^12.0.5",
    "@angular/core": "^12.0.5",
    "@angular/router": "^12.0.5",
    "@ngrx/store": "^12.1.0",
    "@spartacus/checkout": "4.1.0-next.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "@spartacus/tracking": "4.1.0-next.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./src/schematics/collection.json"
}
