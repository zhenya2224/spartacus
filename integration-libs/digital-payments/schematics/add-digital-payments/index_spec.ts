/// <reference types="jest" />

import { RunSchematicTaskOptions } from '@angular-devkit/schematics/tasks/run-schematic/options';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  CLI_DIGITAL_PAYMENTS_FEATURE,
  CLI_CHECKOUT_FEATURE,
  LibraryOptions,
  LibraryOptions as SpartacusDigitalPaymentsOptions,
  SpartacusOptions,
  SPARTACUS_CHECKOUT,
  SPARTACUS_SCHEMATICS,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');
const featureModulePath =
  'src/app/spartacus/features/digital-payments/digital-payments-feature.module.ts';

describe('Spartacus Digital-Payments schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: SpartacusDigitalPaymentsOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const digitalPaymentsFeatureOptions: SpartacusDigitalPaymentsOptions = {
    ...libraryNoFeaturesOptions,
    features: [CLI_DIGITAL_PAYMENTS_FEATURE],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      path.join(
        __dirname,
        '../../../../projects/schematics/src/collection.json'
      )
    );
    schematicRunner.registerCollection(
      SPARTACUS_CHECKOUT,
      path.join(
        __dirname,
        '../../../../feature-libs/checkout/schematics/collection.json'
      )
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        SPARTACUS_SCHEMATICS,
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', libraryNoFeaturesOptions, appTree)
        .toPromise();
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(featureModulePath)).toBeFalsy();
    });
  });

  describe('Digital-Payments feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', digitalPaymentsFeatureOptions, appTree)
          .toPromise();
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should run the proper installation tasks', async () => {
        const tasks = schematicRunner.tasks
          .filter((task) => task.name === 'run-schematic')
          .map(
            (task) => task.options as RunSchematicTaskOptions<LibraryOptions>
          );
        expect(tasks.length).toEqual(2);

        const checkoutTask = tasks[0];
        expect(checkoutTask).toBeTruthy();
        expect(checkoutTask.name).toEqual('add-spartacus-library');
        expect(checkoutTask.options).toHaveProperty(
          'collection',
          SPARTACUS_CHECKOUT
        );
        expect(checkoutTask.options.options?.features).toEqual([]);

        const checkoutTaskWithSubFeatures = tasks[1];
        expect(checkoutTaskWithSubFeatures).toBeTruthy();
        expect(checkoutTaskWithSubFeatures.name).toEqual(
          'add-spartacus-library'
        );
        expect(checkoutTaskWithSubFeatures.options).toHaveProperty(
          'collection',
          SPARTACUS_CHECKOUT
        );
        expect(checkoutTaskWithSubFeatures.options.options?.features).toEqual([
          CLI_CHECKOUT_FEATURE,
        ]);
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...digitalPaymentsFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });
});
