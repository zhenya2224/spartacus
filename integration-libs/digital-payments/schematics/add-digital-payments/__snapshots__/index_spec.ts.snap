// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Digital-Payments schematics: ng-add Digital-Payments feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { DigitalPaymentsModule, dpTranslationChunksConfig, dpTranslations } from \\"@spartacus/digital-payments\\";

@NgModule({
  declarations: [],
  imports: [
    DigitalPaymentsModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: dpTranslations,
      chunks: dpTranslationChunksConfig,
    },
  })]
})
export class DigitalPaymentsFeatureModule { }
"
`;
